backend/Gemfile

```diff
@@ -35,25 +35,27 @@ end
 
 group :development do
   # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
-  gem 'web-console', '>= 3.3.0'
-  gem 'listen', '~> 3.2'
+  gem 'web-console', '>= 4.1.0'
+  # Display performance information such as SQL time and flame graphs for each request in your browser.
+  # Can be configured to work on production as well see: https://github.com/MiniProfiler/rack-mini-profiler/blob/master/README.md
+  gem 'rack-mini-profiler', '~> 2.0'
+  gem 'listen', '~> 3.3'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
-  gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
 group :test do
   # Adds support for Capybara system testing and selenium driver
-  gem 'capybara', '>= 2.15'
-  gem 'selenium-webdriver'
+  gem 'capybara', '>= 3.26'
+  gem 'selenium-webdriver', '~> 4.0.0.beta3'
   # Easy installation and use of web drivers to run system tests with browsers
   gem 'webdrivers'
 end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
-gem 'devise', '~> 4.7', '>= 4.7.3'
-gem 'doorkeeper', '~> 5.4'
-gem 'doorkeeper-device_authorization_grant', '~> 0.2.0'
+gem 'devise', '~> 4.8'
+gem 'doorkeeper', '~> 5.5', '>= 5.5.1'
+gem 'doorkeeper-device_authorization_grant', github: 'exop-group/doorkeeper-device_authorization_grant', branch: 'doorkeeper-5_5'
```

Comment:
This is temporary, just waiting for the new release, which will be possible as soon as https://github.com/exop-group/doorkeeper-device_authorization_grant/pull/6 is merged

Reply:
https://github.com/exop-group/doorkeeper-device-flow-example/pull/5/commits/5ad77143cd46d1650583d067cf933afd9d5ba012